### PR TITLE
Make C/C++ API backend agnostic

### PR DESCRIPTION
**Pull request by 9prady9**

### CODE REVIEW

File: include/fg/image.h [hunk 1]
```diff
@@ -114,7 +114,7 @@ FGAPI fg_err fg_get_image_type(fg_dtype* pOut, const fg_image pImage);
 
    \return \ref fg_err error code
  */
-FGAPI fg_err fg_get_image_pbo(unsigned* pOut, const fg_image pImage);
+FGAPI fg_err fg_get_image_resource_id(unsigned* pOut, const fg_image pImage);
```
Comment (pavanky) [inline include/fg/image.h]: Why is this `image_resource_id` as opposed to `image_buffer` ?

File: include/fg/image.h [hunk 2]
```diff
@@ -230,16 +230,16 @@ class Image {
         FGAPI dtype channelType() const;
 
         /**
-           Get the OpenGL Pixel Buffer Object identifier
+           Get the image data resource identifier
```
Comment (pavanky) [inline include/fg/image.h]: Same as above

File: include/fg/image.h (hunk 1 above)
Comment (9prady9) [inline include/fg/image.h]: Correcting it, what should be c++ method name ? @shehzan10 suggested `pixels`

File: include/fg/plot.h [hunk 3]
```diff
@@ -253,58 +253,58 @@ class Plot {
...
         FGAPI unsigned alphas() const;
 
         /**
-           Get the OpenGL buffer object identifier for markers sizes, per vertex
+           Get the buffer identifier for per vertex marker sizes
 
-           \return OpenGL VBO resource id.
+           \return marker sizes buffer resource id.
          */
         FGAPI unsigned markers() const;
```
Comment (pavanky) [inline include/fg/plot.h]: You have `markers()` in C++ API, but the C API calls it `markersize_buffer`. Is this consistent ?

Comment (9prady9) [inline include/fg/plot.h]: I dont think either of them are apt/terse for this use case, any suggestions ?

File: include/fg/image.h (hunk 1 above)
Comment (pavanky) [inline include/fg/image.h]: Change both to pixel_buffer and pixels?

Comment (9prady9) [inline include/fg/image.h]: Okay, changing c version to `pixel_buffer` and c++ as `pixels` similar to how others are 
`vertex_buffer` / `vertices`
 `color_buffer` / `colors` 
 `alpha_buffer` / `alphas` 
are named. is that fine ?

Comment (pavanky) [inline include/fg/image.h]: Sounds fine to me.

File: include/fg/plot.h (hunk 3 above)
Comment (pavanky) [inline include/fg/plot.h]: I am not entirely sure what this is being used for, so I don't know what is appropriate. I just noticed that they were not consistent.

@shehzan10 thoughts ?

**Comment by shehzan10** [comment]
There are a bunch of compilation errors where `markers` have not been changed to `radii`.

**Comment by 9prady9** [comment]
I think i didn't push whatever i changes i made in the night, please check now.

**Pull request merged**